### DISCUSSION TITLE
Does it change the route on load?

### DISCUSSION THREAD

**Discussion by chan-dev** [stephanrauh/ngx-extended-pdf-viewer#902] [Q&A]
For some reason, it seems like it calls something on the router upon load? It only happens when i import this library.

Is it calling something that changes the URL or calling the route?

**Comment by stephanrauh** [comment]
As far as I know, it doesn't. The only plausible culprit is the file `pdf_history.js` of pdf.js. But this class is deactivated in ngx-extended-pdf-viewer. In stand-alone mode, pdf.js adds an entry to the browser history each time you navigate to another page or jump to another chapter of the PDF.

How does the problem show? Can you see anything useful in the stacktrace of the debugger?

**Comment by chan-dev** [comment]
I haven't found any clue to which the culprit is, I just ended up creating an angular element to wrap the library and it no longer changes the URL or route w/c i have no clue how it fixed it.

**Reply by stephanrauh** [reply]
Well, at least you've found a solution. But I'd prefer to solve the problem!

BTW, I wonder if wrapping the PDF viewer as a webcomponent solves a problem. pdf.js pollutes the global name space, so you can't display two pdf viewers simultaneously. Wrapping the viewer in a webcomponent should give each instance it own namespace, shouldn't it?

**Reply by chan-dev** [reply]
I'm not quite sure about this since i'm just learning Angular elements, but yeah i believe wrapping them in web components will scope the functionalities on each instance.

**Reply by stephanrauh** [reply]
THX!
